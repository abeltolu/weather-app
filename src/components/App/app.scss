body {
    margin: 0;
    padding: 0;
    color: #fff;
    font-size: 14px;
    background: rgb(29, 117, 248);
    font-family: 'Mukta', sans-serif;
    overflow: scroll;

    @media (max-width: 768px){
        padding: 20px !important;
    }
}

html, body, main, header, footer, div, a, input, select, p {
    margin: 0;
    padding: 0;
}

.root__app {
    display: flex;
    align-items: center;
    justify-content: center;

    @media screen and (min-width: 768px) and (min-height: 750px){
        height: 100vh;
    }
}

/** BEGIN Weather Error Container CSS **/
.error__container {
    width: 100%;
    padding: 0px;
    font-size: 14px;
}

.error__msg {
    background: #dc1717;
    color: #fff;
    min-height: 10px;
    padding: 10px 20px;
    text-align: center;
    border-radius: 5px;
    text-transform: capitalize;
}
/** END Weather Error Container CSS **/

/** BEGIN Weather App CSS **/
.weather__app__main {
    width: 400px;
    max-width: 400px;
    padding: 20px;
    background: rgb(10, 90, 210);
    display: flex;
    flex-direction: column;
    align-items: center;
    border-radius: 5px;
    margin: 40px auto 40px auto;
    animation: ease-in 5s infinite;

    @media (max-width: 768px){
        width: 100%;
    }
}

.weather__app__container {
    margin: 0 auto;
    width: 100%;
  
    .weather__forecast__tiles {
        width: 100%;
        display: flex;
        flex-direction: column;
    
        .forecast__tile {
            min-height: 10px;
            background: transparent;
            margin: 3px;
            cursor: pointer;
            overflow: hidden;
            position: relative;
    
            .icon {
                width: auto;
                height: auto;
                display: flex;
                align-items: center;
                font-size: 14px;

                & img {
                    width: 50px;
                    height: 50px;
                }
            }

            .min-max {
                padding: 10px 10px;
                font-size: 20px;
                color: #c7cef5;
    
                strong {
                    font-weight: bold;
                    color: #fff;
                }
            }
    
            .tile__main__info {
                display: flex;
                align-items: center;
                justify-content: space-between;
                padding: 0 20px;
                font-size: 14px;

                & .date {
                    text-transform: capitalize;
                }
            }

            :hover {
                background: #256fde;
            }
        }
    }

    .detailed__forecast__info {
        width: 100%;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        margin-top: 20px;

        .city-info {
            width: 100%;
            margin-bottom: 20px;
            display: flex;
            flex-direction: column;
            align-items: center;

            .temp {
                display: flex;
                justify-content: center;

                span.min, span.max, span.sub {
                    font-size: 12px;
                }

                span.sub{
                    top: 0px;
                    left: 80px;
                    width: 100px;
                    display: flex;
                    font-weight: 400;
                    position: absolute;
                }

                span.minmax{
                    margin-left: 20px;
                }

                span.main {
                    font-size: 80px;
                    font-weight: 700;
                    position: relative;
                    margin: 0;
                    line-height: 1;
                }
            }

            .city-name {
                margin: 0;
                font-size: 14px;
                font-weight: 400;
            }
        }
        
        .icon {
            img {
                width: 50px;
            }
        }

        .forecast__hourly__view {
            width: 100%;
            display: flex;
            flex-wrap: wrap;
            padding: 0;
            min-height: 50px;
            margin-bottom: 20px;
            -webkit-transition: height 0.20s ease-in;
            -moz-transition: height 0.20s ease-in;
            -o-transition: height 0.20s ease-in;
            transition: height 0.20s ease-in;
            background: #256fde;
            padding: 10px 0;

            .forecast__hourly__info {
                display: flex;
                flex-direction: column;
                flex: 1;
                align-items: center;
                justify-content: center;
                overflow-x: scroll;

                .hour__temperature {
                    font-size: 16px;
                    font-weight: bold;
                }

                .hour__of__the__day {
                    font-weight: normal;
                    font-size: 12px;
                    color: #c5cae9;
                }

                .hour__cloud__picture {
                    line-height: 0;
                    img {
                        height: 30px;
                    }
                }

                div {
                    padding: 0;
                }
            }
        }
    }

    @media only screen and (max-width: 64em) {
        width: 100%;
    }
    
    @media only screen and (max-width: 48em) {
        width: 100%;
    }
}
/** END Weather App CSS **/