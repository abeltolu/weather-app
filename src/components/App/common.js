import styled from 'styled-components';

export const Header = styled.header`
font-size: 20px;
color: #fff;
margin-bottom: 20px;
`;

export const InputContainer = styled.div`
    width: 100%;
    margin-bottom: 20px;
    display: flex;
    border-radius: 5px;
`;

export const Input = styled.input`
flex: 1;
outline: 0;
border: none;
font-size: 14px;
height: 40px;
padding: 0 10px;
text-transform: capitalize;
border-top-left-radius: 5px;
border-bottom-left-radius: 5px;
`;

export const Button = styled.button`
outline: none;
border: none;
cursor: pointer;
font-size: 14px;
padding: 0 20px;
color: #fff;
background-color: #333;
border-top-right-radius: 5px;
border-bottom-right-radius: 5px;

:hover {
    background: red;
}
`;