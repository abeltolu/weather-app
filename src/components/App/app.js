import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import Loader from 'react-loader-spinner'
import { Header } from './common';
import CityInput from '../Weather/CityInput/cityInputContainer';
import WeatherForecast from '../Weather/Forecast';
import './app.scss';
import './weathericons.scss';
export default class App extends React.Component {
    render() {
        const { status, forecast, isLoading } = this.props;
        return (
            <Fragment>

                <div className="weather__app__main">

                    {/** App Header */}
                    <Header className="main__header">
                        <i className="weathericons wi-day-cloudy"></i> Weather App
                    </Header>
                    
                    {/** Input Field for city */}
                    <CityInput />

                    {
                        //Check if data is loading
                        (isLoading) ?
                        (
                            <Loader type="Ball-Triangle" color="#FFFFFF" height="50" width="50" />
                        )
                        :
                        (
                            /** Check if weather forecasts data is available */
                            (status !== null && status == "failed") ? 
                            (
                                <div className="error__container">
                                    <div className="error__msg">
                                        { (forecast && forecast.message) ? forecast.message : 'City not found' }
                                    </div>
                                </div>
                            ) : (
                                status=="success" && <WeatherForecast forecast={forecast} />
                            )
                        )
                    }
                </div>
            </Fragment>
        );
    }
}

App.propTypes = {
    status: PropTypes.string,
    forecast: PropTypes.object,
    isLoading: PropTypes.bool
}