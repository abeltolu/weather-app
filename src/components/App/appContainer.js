import { connect } from 'react-redux';
import App from './app';

function mapStateToProps(state){
    return {
        forecast: state.weather.forecast,
        status: state.weather.status,
        isLoading: state.weather.isLoading
    };
}

function mapDispatchToProps(dispatch){
    return {}
}

export default connect(mapStateToProps, mapDispatchToProps)(App);