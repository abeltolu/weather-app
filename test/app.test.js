import React from "react";
import renderer from 'react-test-renderer';
import { render, mount, shallow } from "enzyme";

import App from "../src/components/App/app";

describe("<App />", () => {
    
    it("should render snapshot", () => {
        const wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot();
    });

    it("header should contain *Weather App*", () => {
        const wrapper = shallow(<App />);
        const header = wrapper.find('.weather__app__main');
        expect(header.text()).toContain('Weather');
    });

    it("should receive status prop", () => {
        const wrapper = shallow(<App />);
        expect(wrapper.prop('status')).toBeUndefined();
    });

    it("should return city not found", () => {
        const wrapper = shallow(<App status="failed" />);
        const errorContainer = wrapper.find('.error__msg');
        expect(errorContainer.text()).toEqual('City not found');
    });
});