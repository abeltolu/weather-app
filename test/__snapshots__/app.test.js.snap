// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render snapshot 1`] = `
<Fragment>
  <div
    className="weather__app__main"
  >
    <styled.header
      className="main__header"
    >
      <i
        className="weathericons wi-day-cloudy"
      />
       Weather App
    </styled.header>
    <Connect(CityInput) />
  </div>
</Fragment>
`;
